torch/_inductor/triton_helpers.py

```diff
@@ -28,14 +37,35 @@ def maximum(a, b):
     return tl.where(mask, a, b)
 
 
-@triton.jit
-def min(a, dim):
-    return tl.reduce(a, dim, minimum)
+if TRITON_HAS_REDUCE:
 
+    @triton.jit
+    def min2(a, dim):
+        return tl.reduce(a, dim, minimum)
 
-@triton.jit
-def max(a, dim):
-    return tl.reduce(a, dim, maximum)
+    @triton.jit
+    def max2(a, dim):
+        return tl.reduce(a, dim, maximum)
+
+else:
+
+    @triton.jit
+    def min2(a, dim):
+        min_values = tl.min(a, dim)
+        if is_floating(a):
+            has_nan = tl.sum(a != a, dim)
+            nan = tl.full([1], float("nan"), tl.float32).to(a.dtype)
+            min_values = tl.where(has_nan, nan, min_values)
```

Comment:
This is an alternative implementation not using `tl.reduce` while we wait for everyone to support it.